cyclonedx-lib/src/temurin/sbom/TemurinGenSBOM.java

```diff
@@ -18,52 +17,223 @@
+                name = args[++i];
+            } else if (args[i].equals("--value")) {
+                value = args[++i];
+            } else if (args[i].equals("--url")) {
+                url = args[++i];
+            } else if (args[i].equals("--comment")) {
+                comment = args[++i];
+            } else if (args[i].equals("--hashes")) {
+                hashes = args[++i];
+            } else if (args[i].equals("--createNewSBOM")) {
+                cmd = "createNewSBOM";
+            } else if (args[i].equals("--addMetadata")) {        // Metadata Component. We can set "name" for Metadata->Component.
+                cmd = "addMetadata";
+            } else if (args[i].equals("--addMetadataProp")) {    // MetaData Component --> Property -> name-value: os, arch, variant, scmRef, buildRef, full_version_output, makejdk_any_platform_args
+                cmd = "addMetadataProperty";
+            } else if (args[i].equals("--addComponent")) {       // Components->Property: will add name-value.
+                cmd = "addComponent";
+            } else if (args[i].equals("--addComponentProp")) {  // Components->Property: will add name-value.
+                cmd = "addComponentProp";
+            } else if (args[i].equals("--addExternalReference")) {
+                cmd = "addExternalReference";
+            } else if (args[i].equals("--addComponentExtRef")) {
+                cmd = "addComponentExternalReference";
+            }
         }
-        System.out.println("");
+        switch (cmd) {
+            case "createNewSBOM":                            // Creates JSON file
+                Bom bom = createBom(name, version);
+                writeJSONfile(bom, fileName);
+                break;
 
-        Bom bom = createTestBom();
-        String json = generateBomJson(bom);
+            case "addMetadata":                              // Adds Metadata Component --> name
+                bom = addMetadata(name, fileName);
+                writeJSONfile(bom, fileName);
+                break;
 
-        System.out.println("SBOM: " + json);
+            case "addMetadataProperty":                     // Adds MetaData--> Component --> Property --> name-value:
+                bom = addMetadataProperty(name, value, fileName);
+                writeJSONfile(bom, fileName);
+                break;
+
+            case "addComponent":                            // Adds Component
+                bom = addComponent(name, fileName);
+                writeJSONfile(bom, fileName);
+                break;
+
+            case "addComponentProp":                       // Adds Components --> name-value pairs
+                bom = addComponentProperty(name, value, fileName);
+                writeJSONfile(bom, fileName);
+                break;
+
+            case "addExternalReference":                                     // Adds external Reference
+                bom = addExternalReference(url, hashes, comment, fileName);
```

Comment:
url, hashes params are in different order to the addExternalReference() method below...